app/models/concerns/sentiment_analysis_helper.rb

```diff
@@ -0,0 +1,40 @@
+module SentimentAnalysisHelper
+  extend ActiveSupport::Concern
+
+  def opening_sentiments
+    records = incoming_messages.first(average_message_count)
+    average_sentiment(records)
+  end
+
+  def closing_sentiments
```

Comment:
We should be showing closing sentiments only if the conversation is resolved. Should we check that too

Reply:
@scmmishra Yes, I have not added the calls from the controller yet or in the `push_event_data` for the conversation. I was thinking, I will add the condition there itself.